### DISCUSSION TITLE
[Question] 是否支持构建AppleTVSimulator,iPhoneSimulator,WatchSimulator

### DISCUSSION THREAD

**Discussion by leleliu008** [xmake-io/xmake#2184] [General]
我看到xmake支持 `MacOSX` `AppleTVOS` `WatchOS` `iPhoneOS`，没有看到写有支持`AppleTVSimulator`, `iPhoneSimulator`, `WatchSimulator`，想确认一下，是否支持构建Simulator的，如果支持，使用哪个参数？谢谢

**Comment by waruqi** [comment]
支持，直接切 arch 到 x86_64就行了

**Reply by leleliu008** [reply]
谢谢，我刚刚下载源码看到了，在CHANGELOG.md写的一句话`合并iphoneos和iphonesimulator平台，以及watchos和watchsimulator平台，通过arch来区分，使得打包更加方便，能够支持一次性打包iphoneos的所有arch到一个包中`，这个实现不对吧？因为Simulator也支持arm64，如何区分到底是iphoneos还是iphonesimulator呢？

**Reply by leleliu008** [reply]
是不是应该增加一个参数，比如--simulator之类的参数进行区分到底是不是需要构建模拟器平台，仅仅靠-a/--arch=ARCH参数是区分不了的。

**Reply by leleliu008** [reply]
我看实现代码里没有支持arm64的simulator，Apple现在是支持arm64的simulator的，Apple现在全面支持arm，而且新的xcode已经不支持32位的了，这个实现可能需要更新了。

**Reply by waruqi** [reply]
不行就只能再拆分开来了 你可以提个 issues 过来

**Reply by leleliu008** [reply]
刚刚提了3个issue

另外，我写了Android和iOS的github actions workflow，你是否接受pr，如果接受，我可以提交一下

iOS：       https://github.com/leleliu008/test/actions/runs/2013694383
Android：https://github.com/leleliu008/test/actions/runs/2013195105

这是我经过测试的

**Reply by waruqi** [reply]
不用了，目前的 ci jobs 已经足够，再多影响效率。

**Reply by waruqi** [reply]
另外，安装 xmake 有专门的 action 干嘛不用，没必要自己编译：https://github.com/xmake-io/github-action-setup-xmake

**Reply by leleliu008** [reply]
我是自己写package manager的，所以自己编译

**Reply by leleliu008** [reply]
因为有几个项目是用xmake作为构建系统的，所以就研究了一下，发现按照文档上说的编译为Android和iOS行不通，必需加上--toolchain参数才行，但是文档里没有写。所以，我看了你的github actions里没有关于Android和iOS的workflow，我觉得你可能没有做大量测试，我觉得大量测试是有必要的

**Reply by waruqi** [reply]
xmake 自身在 ios android 上跑，属于冷门场景，很少会用户这上面跑，暂时就不考虑加 ci 了，有问题直接改，否则太浪费 ci 资源， 反而影响日常开发效率

如果是 ios android 编译测试，xmake-repo 上原本就有 ios android ci压测。
